dashboard/test/lib/resources_autocomplete_test.rb

```diff
@@ -6,29 +6,39 @@ class ResourcesAutocompleteTest < ActiveSupport::TestCase
   # as fixtures. These are defined in test/fixtures/resource.yml
 
   test "finds resource with matching name" do
-    matches = ResourcesAutocomplete.get_search_matches('Studi', 5)
+    matches = ResourcesAutocomplete.get_search_matches('Studi', 5, nil)
     assert_equal 1, matches.length
     assert_equal 'Code Studio', matches[0]['name']
   end
 
   test "finds resource with matching url" do
-    matches = ResourcesAutocomplete.get_search_matches("wikip", 5)
+    matches = ResourcesAutocomplete.get_search_matches("wikip", 5, nil)
     assert_equal 1, matches.length
     assert_equal 'wiki', matches[0]['name']
   end
 
   test "finds multiple matches" do
-    matches = ResourcesAutocomplete.get_search_matches("class", 5)
+    matches = ResourcesAutocomplete.get_search_matches("class", 5, nil)
     assert_equal 2, matches.length
   end
 
+  test "limits matches by course version id" do
+    matches = ResourcesAutocomplete.get_search_matches("class", 5, nil)
+    assert_equal 2, matches.length
+
+    # Check that specifying a course version id finds an associated resource
+    course_version_id = CourseVersion.find_by_key('2019').id
+    matches = ResourcesAutocomplete.get_search_matches("class", 5, course_version_id)
+    assert_equal 1, matches.length
+  end
```

Comment:
when a course version id is NOT specified, it looks like we included resources which do have course version ids in the search results. would it be better to only include resources with `nil` course version id in this case?  This is a genuine question :-) I haven't thought through what the rest of the path would look like to get to the place where all resources have course version ids, and as long as we plan to get into that state before we do our final data import + launch, then I could be convinced that it doesn't really matter what we do here. 

In any case, I think it's fine to proceed with this PR as-is, but also think it would be great if you could wrap your head around what the rest of the changes in this space might look like between now and launch at some point.

Reply:
I think when a course version id is not specified, it only includes resources that also do not have a course version id. I'll improve the comments here, but I added a third resource that matches this string for this test. Two match when course_version_id is nil and one matches when it isn't.

Reply:
OK, this sounds fine for now. Stepping back a bit, if the notes I captured from a recent discussion are correct, then the case where a resource does not have a course version id should be going away entirely for January launch: https://codedotorg.atlassian.net/browse/PLAT-505?focusedCommentId=17406&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17406

Reply:
Yep, so in reality 
> I could be convinced that it doesn't really matter what we do here

is the correct behavior :)